src/nomad_simulations/schema_packages/model_system.py

```diff
@@ -1127,9 +1150,78 @@ def resolve_system_type_and_dimensionality(
 
         return system_type, dimensionality
 
+    def _walk_depth_first(self):
+        """Yield self and every descendant ModelSystem (depth-first)."""
+        yield self
+        for sub in self.sub_systems:
+            yield from sub._walk_depth_first()
+
+    def _consolidate_particle_states(self, logger):
+        """
+        Make sure the representative (top-level) ModelSystem owns a complete,
```

Comment:
Mmm, as far as I see in the logic, this is copy-pasting the info of the `ParticleState` sections might exists in the children of the global ModelSystem parent (from L 1187 to 1210)

However, this might explode the archive with a lot of repetitive data